packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +91,9 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	const file = fileURLToPath(`file:///${ctx.file}`)
+		.replace(fileURLToPath(config.root), '/')
+		.replace(/\\/g, '/');
```

Comment:
Thanks for bearing with me so far. This looks great on Windows now!

There is one minor blemish on linux.
<img width="600" alt="image" src="https://github.com/withastro/astro/assets/69170106/02a93381-eb2f-4bcd-b1ce-0e2ba50f246e">

Reply:
Do you think the following could be acceptable?
```ts
	const file = fileURLToPath(`file:///${ctx.file.substring(ctx.file.startsWith('/') ? 1 : 0)}`)
		.replace(fileURLToPath(config.root), '/')
		.replace(/\\/g, '/');
```

Reply:
It could, but this is becoming kinda long. Maybe if the approach from my other comment does not work.